foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/PoetryMigration.java

```diff
@@ -0,0 +1,118 @@
+package com.boozallen.aissemble.upgrade.migration;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.boozallen.aissemble.upgrade.util.FileUtils;
+import com.electronwill.nightconfig.core.Config;
+import com.electronwill.nightconfig.core.file.FileConfig;
+
+/**
+ * 
+ * Baton migration used to migrate aiSSEMBLE python packages to the new naming convention
+ */
+public class PoetryMigration extends AbstractAissembleMigration {
+
+    private static final Logger logger = LoggerFactory.getLogger(PoetryMigration.class);
+    protected static final String DEPENDENCIES_KEY = "tool.poetry.dependencies";
+    protected static final String PYTHON_PACKAGE_PREFIX = "aissemble-";
+    protected static final String UPGRADE_VERSION = "1.7.0.*";
```

Comment:
S: We want to avoid hardcoding this version in if possible. If this release ends up being cut as `1.7.1` then this will be incorrect. Otherwise we have to remember to change this value after each failed release.

You could read in the version from the pom file (see `extensions/extensions-encryption/extensions-encryption-vault-java/pom.xml` for an example). Then based on that read in version, construct the `UPGRADE_VERSION`.

Reply:
This feature has been added.